src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateNew = JSON.parse(JSON.stringify(state));
+
+  for (let key = 0; key < actions.length; ++key) {
+    if (actions[key].type === 'addProperties') {
+      const stateNew1 = Object.assign({}, stateNew, actions[key].extraData);
```

Comment:
[NAMING] - Use more descriptive names for object copies. Instead of `stateNew1`, consider using a name like `stateAfterAddition` to reflect the transformation that has occurred.